packages/public-api/src/packageInfo/packageSummaryService.ts

```diff
@@ -0,0 +1,73 @@
+import { getRepository } from 'typeorm';
+import {
+  PackageMetadata,
+  PackageUsageByHostnameProjection,
+  PackagePopularityView,
+  PackageVulnerability,
+  toSerializable,
+} from '@gradejs-public/shared';
+import semver from 'semver';
+
+export async function getPackageSummaryByName(packageName: string) {
+  const packageRepo = getRepository(PackageMetadata);
+  const packageUsageRepo = getRepository(PackageUsageByHostnameProjection);
+  const packagePopularityRepo = getRepository(PackagePopularityView);
+  const packageVulnerabilities = getRepository(PackageVulnerability);
+
+  const queries = [
+    packageRepo.createQueryBuilder().where('name = :packageName', { packageName }),
+    packageUsageRepo
+      .createQueryBuilder('usage')
+      .leftJoinAndSelect('usage.hostname', 'hostname')
+      .where('package_name = :packageName', { packageName })
+      .limit(16), // TODO: remove hardcode
+    packagePopularityRepo
+      .createQueryBuilder()
+      .where('package_name = :packageName', { packageName }),
+    packageVulnerabilities
+      .createQueryBuilder()
+      .where('package_name = :packageName', { packageName }),
+  ];
+
+  const [[packageInfo], usage, [popularity], vulnerabilities] = (await Promise.all(
+    queries.map((q) => q.getMany())
+  )) as [
+    PackageMetadata[],
+    PackageUsageByHostnameProjection[],
+    PackagePopularityView[],
+    PackageVulnerability[]
+  ];
+
+  if (!packageInfo) {
+    return null;
+  }
+
+  return {
+    ...toSerializable(packageInfo),
```

Comment:
Don't we want to call the `toSerializable` at the exact moment API returns the response, like how it's done across the codebase? Also applies to same calls below.

Reply:
Unfortunately, toSerializable is not recursive, so it doesn't replace inner instances of entity objects' types. Reverting the fix.